### PR TITLE
PMM-10340: simpler k8s registration

### PR DESCRIPTION
**Pull request by solovevayaroslavna**
https://jira.percona.com/browse/PMM-10340
related PR from core-ui - https://github.com/percona-platform/core-ui/pull/70
image
video:
[DBaas Paste from clipboard.webm](https://user-images.githubusercontent.com/90199600/181742543-579b1ef1-d9ba-40d7-bb56-0d30f5093a04.webm)

### CODE REVIEW

File: public/app/percona/dbaas/components/Kubernetes/AddKubernetesModal/AddKubernatesModal.utils.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import { parse } from 'yaml';
+
+export const onKubeConfigValueChange = (value: string | undefined, updateFormMutator: any) => {
```
Comment (fabio-silva) [inline public/app/percona/dbaas/components/Kubernetes/AddKubernetesModal/AddKubernatesModal.utils.ts:3]: Can we add some typings here? I guess it's just a function

File: public/app/percona/dbaas/components/Kubernetes/AddKubernetesModal/AddKubernatesModal.utils.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  updateFormMutator(value, defaultName);
+};
+
+const getClusterNameFromKubeConfig = (value: string | undefined): string | undefined => {
+  if (value) {
+    try {
+      const parsedYAML = parse(value);
+      const clusters = parsedYAML?.clusters;
+      if (clusters && clusters.length) {
+        const clusterWithName = clusters.find((item: any) => item?.name != null);
```
Comment (fabio-silva) [inline public/app/percona/dbaas/components/Kubernetes/AddKubernetesModal/AddKubernatesModal.utils.ts:14]: Does `cluster` have any type?

**Review by fabio-silva — CHANGES_REQUESTED** [review]

File: public/app/percona/dbaas/components/Kubernetes/AddKubernetesModal/AddKubernatesModal.utils.test.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import kubeConfigFile from './KubeConfigTestMock';
+import { onKubeConfigValueChange } from './AddKubernatesModal.utils';
+
+describe('AddKubernetesModal.utils::', () => {
+  it('getClusterNameFromKubeConfig returns name of cluster', () => {
+    const mutatorMock = jest.fn();
+    onKubeConfigValueChange(kubeConfigFile, mutatorMock);
+    expect(mutatorMock).toBeCalledWith(expect.anything(), 'minikube');
```
[the remark this answers is not in the document]
Comment (solovevayaroslavna) [reply public/app/percona/dbaas/components/Kubernetes/AddKubernetesModal/AddKubernatesModal.utils.test.ts]: thanks! I fixed it

File: public/app/percona/dbaas/components/Kubernetes/AddKubernetesModal/AddKubernatesModal.utils.ts [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  updateFormMutator(value, defaultName);
+};
+
+const getClusterNameFromKubeConfig = (value: string | undefined): string | undefined => {
+  if (value) {
+    try {
+      const parsedYAML = parse(value);
+      const clusters: KubeConfig[] = parsedYAML?.clusters;
+      if (clusters && clusters.length) {
+        const clusterWithName = clusters.find((item: any) => item?.name != null);
```
Comment (fabio-silva) [inline public/app/percona/dbaas/components/Kubernetes/AddKubernetesModal/AddKubernatesModal.utils.ts:18]: I see you added typings, please don't forget to remove that `any` from find.

**Review by fabio-silva — CHANGES_REQUESTED** [review]

File: public/app/percona/dbaas/components/Kubernetes/AddKubernetesModal/AddKubernatesModal.utils.ts (hunk 4 above)
Comment (solovevayaroslavna) [reply public/app/percona/dbaas/components/Kubernetes/AddKubernetesModal/AddKubernatesModal.utils.ts:18]: sorry, lost it, I'm going to check all any right now

**Review by matejkubinec — APPROVED** [review]

**Review by fabio-silva — APPROVED** [review]

**Pull request merged**